sdk/go/common/apitype/templates.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016-2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apitype
+
+// A pulumi template remote where the source URL contains
+// a valid Pulumi.yaml file.
+type PulumiTemplateRemote struct {
+	ProjectTemplate
+	Name        string `json:"name"`       // The name of the template
+	SourceName  string `json:"sourceName"` // The name of the template source - for display purposes
+	TemplateURL string `json:"sourceURL"`  // The unique url that identifies the template to the service
+	Runtime     string `json:"runtime"`    // The runtime of the template
+}
+
+type ListOrgTemplatesResponse struct {
+	Templates map[string][]*PulumiTemplateRemote `json:"templates"`
+	// OrgHasTemplates is true len(Templates) > 0
+	OrgHasTemplates bool `json:"orgHasTemplates"`
+	// HasAccessError indicates that the Pulumi service was not able to access a
+	// template source in the org.
+	HasAccessError bool `json:"hasAccessError"`
+	// HasUpstreamError indicates that there was an unspecified error fetching a
+	// template.
+	HasUpstreamError bool `json:"hasUpstreamError"`
```

Comment:
Will these be treated differently on the user side?  It's not quite clear to me what the distinction in these is.  Would it be better to return an error string here instead of two booleans?

Reply:
This API is already in use in the Pulumi service. We can't change these types without changing the existing API, and I don't want to do that (it's live - so changing would be multiple releases).

Reply:
My main worry is that down the line we're gonna need more error "types", and it's gonna be messy on the client side.  And while the API has no users would be the best time to change that still.  Once it's used in `pulumi/pulumi` it's basically impossible to change it.